packages/react-devtools-extensions/src/backend.js

```diff
@@ -14,6 +16,21 @@ function welcome(event) {
     return;
   }
 
+  // In some circumstances, this method is called more than once for a single welcome message.
+  // The exact circumstances of this are unclear, though it may have to do with 3rd part script event batching.
+  // Regardless, this can cause DevTools to add duplicate elements to the Store (and throw an error)
+  // or worse yet, choke up entirely and freeze the browser.
+  // The simplest solution is to ignore the duplicate events.
+  // See https://github.com/facebook/react/issues/24162
+  if (welcomeHasInitialized) {
+    console.warn(
+      'React DevTools detected duplicate welcome "message" events from the content script.',
+    );
+    return;
+  }
+
+  welcomeHasInitialized = true;
+
   window.removeEventListener('message', welcome);
```

Comment:
The above change seems kind of silly, since we're removing the event handler here– but it seems like the batching (mentioned in the comment above) prevents the event handler from being removed properly.

Reply:
It looks like maybe the `removeEventHandler` part of the event handler API is being broken– such that we can _register_ an event handler on the `window` but then never remove it. This is a pretty serious bug that should be fixed (somewhere in react-easy-state maybe?)